collector/lib/runtime-control/Config.h

```diff
@@ -0,0 +1,54 @@
+#ifndef _RUNTIME_CONTROL_CONFIG_H_
+#define _RUNTIME_CONTROL_CONFIG_H_
+
+#include <chrono>
+#include <condition_variable>
+#include <mutex>
+#include <optional>
+
+#include <storage/runtime_filters.pb.h>
+
+#include "Hash.h"
+
+#include "../ResourceSelector.h"
+
+namespace collector::runtime_control {
+
+class Config {
+ public:
+  Config() {
+    for (int i = 0; i < storage::RuntimeFilterFeatures_ARRAYSIZE; i++) {
+      kFeatureFlags_[i] = 1UL << (i + 1);
+    }
+  }
+
+  static Config& GetOrCreate();
+
+  // returns true when initialized, false for a timeout.
+  bool WaitUntilInitialized(unsigned int timeout_ms);
+
+  void Update(const storage::RuntimeFilteringConfiguration& msg);
+  bool IsFeatureEnabled(uint64_t bitMask, storage::RuntimeFilterFeatures feature);
+  bool IsFeatureEnabled(std::string cluster, std::string ns, std::string container_id, storage::RuntimeFilterFeatures feature);
+  bool IsFeatureEnabled(std::string cluster, std::string ns, storage::RuntimeFilterFeatures feature);
+  void ConfigMessageToConfig(const storage::RuntimeFilteringConfiguration& msg);
+
+ private:
+  std::mutex mutex_;
+  std::condition_variable condition_;
+  std::optional<storage::RuntimeFilteringConfiguration> config_message_;
+  uint64_t kFeatureFlags_[storage::RuntimeFilterFeatures_ARRAYSIZE];
+  UnorderedMap<std::string, uint64_t> namespaceFeatureBitMask_;
+  UnorderedMap<std::string, uint64_t*> containerFeatureBitMask_;
+  uint64_t SetFeatureBitMask(uint64_t bitMask, bool enabled, storage::RuntimeFilterFeatures feature);
+  void SetBitMask(std::string cluster, std::string ns);
+  void SetBitMask(std::string cluster, std::string container_id, std::string ns);
+  void SetBitMasksForNamespaces(std::string cluster);
```

Comment:
These also don't seem to need to make copies of every single argument, at best they should create a copy when they do the final storing on a member of this class.

Reply:
See above comments. This is fixed.